### ISSUE TITLE
middleware/etcd: tests are flaky

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#207]
The tests in the etcd middleware are extremely flaky:

```
% go test -tags=etcd -v -run='TestLookup'
=== RUN   TestLookup
--- PASS: TestLookup (8.51s)
PASS
ok      github.com/miekg/coredns/middleware/etcd    8.515s
```

```
% go test -tags=etcd -v -run='(TestLookup|TestProxyLookupFailDebug)'
=== RUN   TestProxyLookupFailDebug
--- PASS: TestProxyLookupFailDebug (5.00s)
=== RUN   TestLookup
--- PASS: TestLookup (5.45s)
PASS
ok      github.com/miekg/coredns/middleware/etcd    10.462s
```

And... now it works (it did loose some internet connectivity in when debugging this). Would no external access impact the tests? (Need to look into that).

Example of a failing test in travis: https://travis-ci.org/miekg/coredns/jobs/150829467

**Comment by miekg** [comment]
Debugging this and other failures on a hotel network... my DNS is intercepted leading to funky answers (because it goes out to the internet). This is annoying and means I have to setup a complete tree of servers to properly test things and make all queries local to the machine.

**Comment by miekg** [comment]
Fixed by #213 and #214
